### ISSUE TITLE
[$250] Settings - Profile - 'Magic sign-in code' mail instead of 'Verify secondary email' mail

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#44727]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.3-5
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** albertodege+2024@gmail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
Prerequisite: The user is logged in to the website.
1. Navigate to account Settings > Profile
2. Click on "Contact Method"
3. Click on "New Contact method"
4. Enter an email into the new contact field
5. Click on "Add"
6. Observe that you're redirected to the "Contact methods" page
7. Click on the new contact method
8. Click on "Resend Magic code"

## Expected Result:
The user should receive the 'Verify secondary email' email again.

## Actual Result:
The user receives the 'Magic sign-in code' email. It looks like the user should be logging in to Expensify instead of validating their account.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/9bb2d944-d0c7-4a72-afd1-86a35a5ff578

![Bug6517879_1718754148044!Expensify_magic_sign-in_code_mail](https://github.com/Expensify/App/assets/115492554/a7bd3777-bcc8-47cb-8b83-344b539385df)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013c373cc71448a82e</li>
        <li>Upwork Job ID: 1808491110713348343</li>
        <li>Last Price Increase: 2024-07-10</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mkhutornyi</details>

**Comment by dangrous** [comment]
@abekkala - would love your thoughts here. I think this **is** a backend issue, but I'm not sure it's worth fixing - It's a pretty minor difference, and might take a bit of time to unravel.

Also @izarutskaya - was this discovered as part of a regression test (i.e. was it different until recently), or just a generic bug report? Thanks!

**Comment by dangrous** [comment]
bumping the above when y'all get a chance - thank you!

**Comment by abekkala** [comment]
@dangrous 
> @abekkala - would love your thoughts here. I think this is a backend issue, but I'm not sure it's worth fixing - It's a pretty minor difference, and might take a bit of time to unravel.

Oh, if it's backend but not a high priority right now we can close. 
Does the different email actually affect their ability to validate? Is it just the wording of the email is wrong or are they unable to actually validate?

**Comment by abekkala** [comment]
we've determined this will be an internal fix but still determining the priority

**Comment by dangrous** [comment]
> Does the different email actually affect their ability to validate? Is it just the wording of the email is wrong or are they unable to actually validate?

Just the wording - the link works either way

**Comment by abekkala** [comment]
@dangrous ok, got it. 

I think it's something that _should_ get fixed as it does come across confusing for the end user, but understand the lower priority given it does _work_

Are you in favor of closing or changing the immediacy label to weekly, monthly? Or Do you think it should be a 'hot pick' for someone else to pick up who might have the bandwidth?

**Comment by dangrous** [comment]
Will maybe be able to get to this next week? Not looking likely this week

**Comment by dangrous** [comment]
put up some code that shoudl fix this

**Comment by dangrous** [comment]
Fix is on staging, should be on prod and ready to close out shortly!

**Comment by dangrous** [comment]
Okay I broke something with this so it's being partially reverted, I'll give it another go soon.

https://github.com/Expensify/App/issues/46293#issuecomment-2253191900

**Comment by dangrous** [comment]
Okay we're on prod now. @izarutskaya (or @abekkala) do you mind re-testing to confirm we can close this out? Thanks!

**Comment by dangrous** [comment]
Just tested this, looks good on staging! Received the verify secondary email once when I added the email, and then again when I clicked "Didn't receive a magic code?"

I don't believe we need payment on this one but if I'm mistaken @mkhutornyi let me know!
